### ISSUE TITLE
Bug in expand arc : bounds are copied from an inlet to outlets 

### ISSUE THREAD

**Issue by ReinboldV** [Pyomo/pyomo#893]
I am using pyomo.network to build sort of graph with blocks and extensive flow connections. In this application flow variable are extensive (or conservative, see #882), positive or negative (general case). Block may have inlets and outlets,  counted positive when the flow come respectively in and out. But In some cases I need a flow to be reversible, which means positive or negative some times.  

During the arc expansion, some properties of the ports, like rule, up bound or low bound,  are copied to the expanded arc variables (named $arc_name$_expanded). For Extensive connection, this actually make sense if all the connected variable are positive. Let's suppose their are two sources feeding one load. If the load has boundaries, it makes sense to impose these limits to the expended arc variables. 
But it is actually quiet restrictive when considering positive and negative variables. Let's see an example to illustrate my purpose. Let's have to sources feeding one destination. An the destination has a dummy  bounds rule (lambda m, t : (-10, 10) ).

```
from pyomo.environ import *                                        
from pyomo.network import *                                        
from pyomo.dae import *                                            
                                                                   
m = ConcreteModel()                                                
                                                                   
m.b1 = Block()                                                     
m.b2 = Block()                                                     
m.b4 = Block()                                                     
                                                                                                                               
def source_block(b):                                               
    b.t = ContinuousSet(bounds=(0, 1))                             
    b.p_out = Var(b.t)                                             
    b.outlet = Port(initialize={'p': (b.p_out, Port.Extensive)})   
                                                                   
def load_block(b):                                                 
    b.t = ContinuousSet(bounds=(0, 1))                             
    b.p_in = Var(b.t,  bounds=lambda m, t: (-10, 10))              
    b.inlet = Port(initialize={'p': (b.p_in, Port.Extensive)})     
                                                                   
source_block(m.b1)                                                 
source_block(m.b2)                                                 
load_block(m.b4)                                                   
                                                                   
m.b14 = Arc(source=m.b1.outlet, destination=m.b4.inlet)            
m.b24 = Arc(source=m.b2.outlet, destination=m.b4.inlet)            
                                                                   
TransformationFactory('dae.finite_difference').apply_to(m, nfe=1)  
TransformationFactory("network.expand_arcs").apply_to(m)                                                      
``` 
A pprint() would give us this : 
```
5 Block Declarations
    b1 : Size=1, Index=None, Active=True
        1 Var Declarations
            p_out : Size=2, Index=b1.t
                Key : Lower : Value : Upper : Fixed : Stale : Domain
                  0 :  None :  None :  None : False :  True :  Reals
                  1 :  None :  None :  None : False :  True :  Reals

        1 Constraint Declarations
            outlet_p_outsum : Size=2, Index=b1.t, Active=True
                Key : Lower : Body                            : Upper : Active
                  0 :   0.0 : b14_expanded.p[0] - b1.p_out[0] :   0.0 :   True
                  1 :   0.0 : b14_expanded.p[1] - b1.p_out[1] :   0.0 :   True

        1 ContinuousSet Declarations
            t : Dim=0, Dimen=1, Size=2, Domain=None, Ordered=Sorted, Bounds=(0, 1)
                [0, 1]

        1 Port Declarations
            outlet : Size=1, Index=None
                Key  : Name : Size : Variable
                None :    p :    2 : b1.p_out

        4 Declarations: t p_out outlet outlet_p_outsum
    b14_expanded : Size=1, Index=None, Active=True
        1 Var Declarations
            p : Size=2, Index=b1.t
                Key : Lower : Value : Upper : Fixed : Stale : Domain
                  0 :  None :  None :  None : False :  True :  Reals
                  1 :  None :  None :  None : False :  True :  Reals

        1 Declarations: p
    b2 : Size=1, Index=None, Active=True
        1 Var Declarations
            p_out : Size=2, Index=b2.t
                Key : Lower : Value : Upper : Fixed : Stale : Domain
                  0 :  None :  None :  None : False :  True :  Reals
                  1 :  None :  None :  None : False :  True :  Reals

        1 Constraint Declarations
            outlet_p_outsum : Size=2, Index=b1.t, Active=True
                Key : Lower : Body                            : Upper : Active
                  0 :   0.0 : b24_expanded.p[0] - b2.p_out[0] :   0.0 :   True
                  1 :   0.0 : b24_expanded.p[1] - b2.p_out[1] :   0.0 :   True

        1 ContinuousSet Declarations
            t : Dim=0, Dimen=1, Size=2, Domain=None, Ordered=Sorted, Bounds=(0, 1)
                [0, 1]

        1 Port Declarations
            outlet : Size=1, Index=None
                Key  : Name : Size : Variable
                None :    p :    2 : b2.p_out

        4 Declarations: t p_out outlet outlet_p_outsum
    b24_expanded : Size=1, Index=None, Active=True
        1 Var Declarations
            p : Size=2, Index=b1.t
                Key : Lower : Value : Upper : Fixed : Stale : Domain
                  0 :   -10 :  None :    10 : False :  True :  Reals
                  1 :   -10 :  None :    10 : False :  True :  Reals

        1 Declarations: p
    b4 : Size=1, Index=None, Active=True
        1 Var Declarations
            p_in : Size=2, Index=b4.t
                Key : Lower : Value : Upper : Fixed : Stale : Domain
                  0 :   -10 :  None :    10 : False :  True :  Reals
                  1 :   -10 :  None :    10 : False :  True :  Reals

        1 Constraint Declarations
            inlet_p_insum : Size=2, Index=b1.t, Active=True
                Key : Lower : Body                                               : Upper : Active
                  0 :   0.0 : b14_expanded.p[0] + b24_expanded.p[0] - b4.p_in[0] :   0.0 :   True
                  1 :   0.0 : b14_expanded.p[1] + b24_expanded.p[1] - b4.p_in[1] :   0.0 :   True

        1 ContinuousSet Declarations
            t : Dim=0, Dimen=1, Size=2, Domain=None, Ordered=Sorted, Bounds=(0, 1)
                [0, 1]

        1 Port Declarations
            inlet : Size=1, Index=None
                Key  : Name : Size : Variable
                None :    p :    2 :  b4.p_in

        4 Declarations: t p_in inlet inlet_p_insum

2 Arc Declarations
    b14 : Size=1, Index=None, Active=False
        Key  : Ports                 : Directed : Active
        None : (b1.outlet, b4.inlet) :     True :  False
    b24 : Size=1, Index=None, Active=False
        Key  : Ports                 : Directed : Active
        None : (b2.outlet, b4.inlet) :     True :  False

7 Declarations: b1 b2 b4 b14 b24 b14_expanded b24_expanded
```

The Arc expansion is actually adding constraints on b24_expanded.p (and strangely not on b14_expanded, which might not be a planned behaviour). But this is actually restrictive because this does not allow, for instance :

b1[0] = 25
b2[0] = -15
b4[0] = 25-15 =10  

which was allowed by the modeller in the first place. 

On the other hand, if the modeller wants to define only positive variables (which was the original intend I guess), i.e. b1 >= 0 and b2 >= 0, then the added constraints on b14_extended and b24_extended are actually supernumerary . 

To properly correct that, I will propose a small PR changing the pyomo.network.utils.replicate_var() method and just delete the lines where domains and bounds are copied to the extended block variables.

**Comment by gseastream** [comment]
To answer the weird behavior where `b14_expanded` does not have the bounds that the other expanded block got, yeah that was not intentional. It happens because the Extensive rule which gets called on every port might be called on those ports in any order (deterministic but not sorted), and when Extensive gets called it does both Split and Combine, so for any given port it will create the expanded block variables for both all inlets and all outlets if they do not already exist, and at that moment replicate the bounds of the variable on that given port. I checked, and the expansion procedure calls the Extensive rules in the order of b1, b4, then b2 for this model, which explains why `b14_expanded` has the bounds of `b1.p_out` (i.e. no bounds), and `b24_expanded` has the bounds of `b4.p_in` (i.e. -10 and 10).

@jsiirola was there a reason we copied the domain and bounds for variables we create? The `replicate_var` function is called both in the Extensive rule procedure, and also in the general arc expansion when populating implicit/auto port members with their variables (if the user gave an empty port or an implicit `None` value for one of the members). I could perhaps see a better argument for copying domains on auto vars, although we don't do any checking in the first place to see if domains match for our explicit members, so I'm not sure. If I recall, we only copied domains and bounds in both of the above cases just for good measure, but I can't remember if there was a numerical reason.
